### ISSUE TITLE
problem in Loss function

### ISSUE THREAD

**Issue by BaldrLector** [bj80heyue/One_Shot_Face_Reenactment#6]
@sanweiliti 
Hi, here some problems at loss design.
the original paper says 'Lreconstruct is the L1 loss between the reenactment result and
the input appearance after face contour resampling'. 
And L_id, L_preceptual ware also  computed 
between result and appearance, but in the code, they ware computed between result and ground-truth(the pose guide)

so is there a mistake? 
thanks

**Comment by sanweiliti** [comment]
Hi, 
it is assumed that the face contour resampling does not change identity information as it only changes the face contour a bit, so in the code L_id and L_preceptual are computed between the result and ground-truth. It should be similar with training the model by calculating L_id, L_preceptual between result and input appearance.

**Comment by BaldrLector** [comment]
From original paper say 
> ' instead of
using the same image for both the appearance reference and pose guide, we replace the face
contour of the pose guide in the face parsing map with the face contour of another randomly
sampled person during the training stage, while keeping inner facial parts unchanged, to
encourage the F to ignore shape information as much as possible '

so gound_truth(provide the pose guide) and input appearance have different identity information, 
or you mean we should use an image as appearance reference and pose guide, since CeleA does not give pair data?

**Comment by BaldrLector** [comment]
> Hi,
> it is assumed that the face contour resampling does not change identity information as it only changes the face contour a bit, so in the code L_id and L_preceptual are computed between the result and ground-truth. It should be similar with training the model by calculating L_id, L_preceptual between result and input appearance.

Thank your reply!
That makes sense, the ground truth in the code is appearance after face contour resampling by randomly selected pose-guided.


I still have some problems,

1. how many optimization steps/epochs for the model generate an acceptable result? 
2. Is there special training strategy to train this model? I use same Lr, beta_2 and weights for loss  as paper, but I can not get the result just like the paper give

**Comment by sanweiliti** [comment]
1. the model is trained with batch_size=16 for ~60 epochs, lr=0.0001
2. during training, the input image is masked with only the face area remained, so we later finetune the trained model with full image (with background) for around 2 epochs, with lr=2e-05

**Comment by BaldrLector** [comment]
Thank you reply,  very helpful.

**Comment by linghai06** [comment]
> @sanweiliti
> Hi, here some problems at loss design.
> the original paper says 'Lreconstruct is the L1 loss between the reenactment result and
> the input appearance after face contour resampling'.
> And L_id, L_preceptual ware also computed
> between result and appearance, but in the code, they ware computed between result and ground-truth(the pose guide)
> 
> so is there a mistake?
> thanks

Hi, did you reproduce the results? I'm just confused by the discrepancy between the terms in the paper and the code implementation.
